src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -217,31 +217,31 @@ public virtual void DefineCurrentThread(int threadId)
         public async Task ExecStep(int threadId, ResultClass resultClass = ResultClass.running)
         {
             string command = "-exec-step";
-            await ThreadCmdAsync(command, resultClass, threadId);
+            await ThreadFrameCmdAsync(command, resultClass, threadId, 0);
```

Comment:
Did you see if this works in LLDB? I think we will now pass `--frame`. Not sure if lldb-mi will complain.

Reply:
Looks like it is fine, but now it has the same behavior of stepping based on the top frame of the thread